src/Compiler/LatteToPhpCompiler.php

```diff
@@ -56,6 +60,37 @@ public function compile(Scope $scope, string $templateContent, array $variables,
         return $this->remapLines($phpContent);
     }
 
+    /**
+     * @param Variable[] $variables
+     * @param Component[] $components
+     */
+    public function compileFile(Scope $scope, string $templatePath, array $variables, array $components): string
+    {
+        $templateContent = file_get_contents($templatePath) ?: '';
+        $phpContent = $this->compile($scope, $templateContent, $variables, $components);
```

Comment:
compile should be private then

Reply:
It is used in tests, I would keep it public

Reply:
Then we should create some other service just to store files, compiler just creates php content and that new service will just save it to file.

Reply:
It seems as unnecessary complication for me.